src/metadata/wavpack_handler.cc

```diff
@@ -0,0 +1,169 @@
+
+    wavpack_handler.cc - this file is part of Gerbera.
+
+    Copyright (C) 2022 Gerbera Contributors
+
+    Gerbera is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License version 2
+    as published by the Free Software Foundation.
+
+    Gerbera is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with Gerbera.  If not, see <http://www.gnu.org/licenses/>.
+
+    $Id$
+*/
+
+/// \file wavpack_handler.cc
+
+#ifdef HAVE_WAVPACK
+#include "wavpack_handler.h" // API
+
+#include <wavpack/wavpack.h>
+
+#include "cds_objects.h"
+#include "iohandler/io_handler.h"
+
+static auto fileFormats = std::map<int, std::string_view> {
+    { WP_FORMAT_WAV, "WAV" },
+    { WP_FORMAT_W64, "Wave64" },
+    { WP_FORMAT_CAF, "CAF" },
+    { WP_FORMAT_DFF, "DSDIFF" },
+    { WP_FORMAT_DSF, "DSD" },
+};
+
+static auto metaTags = std::map<std::string_view, metadata_fields_t> {
+    { "title", M_TITLE },
+    { "artist", M_ARTIST },
+    { "album", M_ALBUM },
+    { "comment", M_DESCRIPTION },
+    { "track", M_TRACKNUMBER },
+};
+
+void WavPackHandler::fillMetadata(const std::shared_ptr<CdsObject>& obj)
+{
+    auto item = std::dynamic_pointer_cast<CdsItem>(obj);
+    if (!item)
+        return;
+
+    auto resource = item->getResource(0);
+
+    char error[80];
+    /* WavpackContext * WavpackOpenFileInput (const char *infilename, char *error, int flags, int norm_offset) */
+    WavpackContext* context = WavpackOpenFileInput(item->getLocation().c_str(), error, OPEN_TAGS | OPEN_2CH_MAX | OPEN_DSD_NATIVE, 0);
+    auto nrChannels = WavpackGetNumChannels(context);
+    resource->addAttribute(R_NRAUDIOCHANNELS, fmt::to_string(nrChannels));
+    auto sampleRate = WavpackGetSampleRate(context);
+    resource->addAttribute(R_SAMPLEFREQUENCY, fmt::to_string(sampleRate));
+    auto bitsPerSample = WavpackGetBitsPerSample(context);
+    resource->addAttribute(R_BITS_PER_SAMPLE, fmt::to_string(bitsPerSample));
+    /*
+    int WavpackGetFileFormat (WavpackContext *wpc);
+        Return the file format specified in the call to WavpackSetFileInformation() when the file
+        was created. For all files created prior to WavPack 5 this will be zero (WP_FORMAT_WAV). Introduced
+        in WavPack 5.0.0. The following formats are currently defined in wavpack.h:
+        - WP_FORMAT_WAV: Microsoft Waveform Audio Format, including BWF and RF64
+        - WP_FORMAT_W64: Sony Wave64
+        - WP_FORMAT_CAF: Apple Core Audio
+        - WP_FORMAT_DFF: Philips DSDIFF
+        - WP_FORMAT_DSF: Sony DSD format
+    */
+    auto fileFormat = WavpackGetFileFormat(context);
+    resource->addAttribute(R_VIDEOCODEC, fileFormats[fileFormat].data());
```

Comment:
AUDIO?

Reply:
That's the current hack to get it into the filtering for dlna profiles (that needs an update as well). I didn't want to create a new attribute while your still refactoring them.